### ISSUE TITLE
a11y red-flags: improve use of semantic markup

### ISSUE THREAD

**Issue by candu** [CityofToronto/bdit_flashcrow#685]
**Description**
A couple of items here around heading levels, `<nav>`, and other semantic markup issues - this aims to fix those.

**Acceptance Criteria**
- [ ] don't use `<h1>`:
  - [ ] for the "in beta" top bar (use a plain `<div>` instead!);
  - [ ] in dialogs (use `<h2>` if it applies to the whole page, or the appropriate sub-heading if it applies to a specific part of the page);
- [ ] figure out which heading level should be used for Add Location, and use that;
- [ ] `<nav>` should use `<ul>`, and should be marked up to indicate structure;
- [ ] provide skip link to skip past navigation bar;
- [ ] use `<fieldset>` and `<legend>` to group layers in map legend;
- [ ] remove `role="checkbox"` from checkboxes;
- [ ] "Viewing data from" menu button should use proper elements / roles;
- [ ] figure out what is meant by links not indicating location to screen reader users, and fix that.

**Additional Notes**
There are other semantic / markup issues with specific parts of the site, but those are given their own issues.

**Comment by candu** [comment]
Closing this now that we have our full a11y feedback - we'll create more specific tasks there.
